### PR TITLE
Adds Locations and return statment functions

### PR DESCRIPTION
**Pull request by Scriptkiddi**
Also fixes some bugs

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -27,3 +27,4 @@ nginx_error_log: /var/log/nginx/error.log
 nginx_gzip: "on"
 nginx_gzip_disable: "msie6"
 nginx_gzip_types: "text/plain text/css application/json application/javascript text/xml application/xml application/xml+rss text/javascript"
+nginx_server_names_hash_bucket_size: 64
```
Comment (SlothOfAnarchy) [inline defaults/main.yml]: newline

File: templates/nginx.conf.j2 [hunk 2]
```diff
@@ -38,7 +38,8 @@ http {
     keepalive_timeout {{ nginx_keepalive_timeout -}};
     types_hash_max_size {{ nginx_types_hash_max_size -}};
     server_tokens {{ nginx_server_tokens -}};
-    
+    server_names_hash_bucket_size {{ nginx_server_names_hash_bucket_size -}};
```
Comment (SlothOfAnarchy) [inline templates/nginx.conf.j2]: why the dash? There is no whitespace to remove. Same above btw.

File: templates/nginx.conf.j2 [hunk 3]
```diff
@@ -97,38 +99,35 @@ http {
...
-
             {% endfor %}
+        }
     {% endfor %}
-    
-    }
-    {% if upstreams is defined %}
-    {% for upstream in upstreams %}
+
+    {% for upstream in upstreams | default([]) %}
```
Comment (SlothOfAnarchy) [inline templates/nginx.conf.j2]: please prefix and document `global`, `upstreams`, `maps`.

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: templates/nginx.conf.j2 (hunk 3 above)
Comment (Scriptkiddi) [reply templates/nginx.conf.j2]: done

File: defaults/main.yml (hunk 1 above)
Comment (Scriptkiddi) [reply defaults/main.yml]: done

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**